modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -428,7 +425,22 @@ protected boolean isContentAlwaysEmpty(HttpResponse msg) {
                 })
                 .addLast("aggregator", aggregator);
             if (handlingSettings.compression()) {
-                ch.pipeline().addLast("encoder_compress", new HttpContentCompressor(handlingSettings.compressionLevel()));
+                ch.pipeline()
+                    .addLast(
+                        "encoder_compress",
+                        new HttpContentCompressor(
```

Comment:
It removes the deprecated function, but it also seems to be changing how compression is done. I'm not sure what was the default before, but now it seems we use deflate/gzip. Can this change be alarming for performance? Meaning there should be some sort of review?